new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Helm.java

```diff
@@ -5,86 +5,175 @@
 
 import java.util.HashMap;
 
-import oracle.weblogic.kubernetes.extensions.LoggedTest;
-
-public class Helm implements LoggedTest {
-
-  private String chartName;
-  private String releaseName;
-  private String namespace;
-  private String repoUrl;
-  private HashMap<String, Object> values;
-
-  public Helm chartName(String chartName) {
-    this.chartName = chartName;
-    return this;
-  }
-
-  public Helm releaseName(String releaseName) {
-    this.releaseName = releaseName;
-    return this;
-  }
-
-  public Helm namespace(String namespace) {
-    this.namespace = namespace;
-    return this;
-  }
-
-  public Helm repoUrl(String repoUrl) {
-    this.repoUrl = repoUrl;
-    return this;
-  }
+import oracle.weblogic.kubernetes.utils.ExecCommand;
+import oracle.weblogic.kubernetes.utils.ExecResult;
+
+import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class Helm {
+
+  /**
+   * install helm chart
+   * @param params the parameters to helm as values
+   * @return true on success, false otherwise
+   */
+  public static boolean install(HelmParams params) {
+    String namespace = params.getNamespace();
+
+    // assertions for required parameters
+    assertThat(namespace)
+        .as("make sure namespace is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    assertThat(params.getReleaseName())
+        .as("make sure releaseName is not empty or null")
+        .isNotNull()
+        .isNotEmpty();
+
+    assertThat(params.getRepoUrl() != null && params.getChartName() == null)
+        .as("make sure chart name is not empty or null when repo url is provided")
+        .isFalse();
+
+    assertThat(params.getRepoUrl() == null && params.getChartDir() == null)
```

Comment:
same as previous (combo check)

Reply:
helm install command uses repo url or chart dir, at least one of these must be provided. Both are not required, hence the combination.